x/oracle/types/oracle.go

```diff
@@ -13,19 +14,73 @@ func (m SignedOracleRegistrationVote) ValidateBasic() error {
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
+
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
-	_, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress); err != nil {
 		return err
 	}
-	_, err = sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress); err != nil {
 		return err
 	}
+
 	if len(m.OracleRegistrationVote.EncryptedOraclePrivKey) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "encryptedOraclePrivKey is empty")
 	}
 	return nil
 }
+
+func (m Oracle) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
+		return sdkerrors.Wrapf(err, "oracle address is invalid. address: %s", m.Address)
+	}
+	return nil
+}
+
+func (m OracleRegistration) ValidateBasic() error {
+	if len(m.UniqueId) == 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
+	}
+	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
+		return sdkerrors.Wrapf(err, "oracle address is invalid. address: %s", m.Address)
+	}
+
+	if m.NodePubKey != nil {
```

Comment:
If `m.NodePubKey` is nil, it will pass this validation. So we need to add checks for `m.NodePubKey` not to be nil.